resources/definitions/fdmprinter.def.json

```diff
@@ -908,6 +908,14 @@
                     "value": "top_bottom_pattern",
                     "settable_per_mesh": true
                 },
+                "skin_angles":
+                {
+                    "label": "Top/Bottom Line Directions",
+                    "description": "A list of integer line directions to use when the top/bottom layers use the lines or zig zag pattern. Elements from the list are used sequentially as the layers progress and when the end of the list is reached, it starts at the beginning again. The list items are separated by commas and the whole list is contained in square brackets. Default is an empty list which means use the traditional default angles (45 and 135 degrees).",
+                    "type": "[int]",
+                    "default_value": "[ ]",
```

Comment:
These settings should both only apply for linear patterns. You need something like `"enabled": "skin_pattern != 'concentric'"`

For the infill patterns that the angles don't apply for Concentric infill, Concentric 3D infill and cubicsubdiv.